pkg/api/validation/validation.go

```diff
@@ -2844,10 +2799,16 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	// Clear status
 	oldNode.Status = node.Status
 
+	// update taints
```

Comment:
I think there are lots of aspects of the pod spec that fall into the "my pod could keep running even though the world has changed if I could just tweak this one aspect", and that it's a slippery slope to make manual fixup of pods a goal.